README.md

```diff
@@ -1,17 +1,265 @@
+   )
+   ```
+
+3. Use the service.
+   - Update the review settings based on application events (using the tracking extension methods).
+      ``` cs
+      using ReviewService;
+
+      private readonly IReviewService<ReviewSettings> _reviewService;
+
+      public async Task DoPrimaryAction(CancellationToken ct)
+      {
+          // Do Primary Action.
+
+          // Track this action.
+          await _reviewService.TrackPrimaryActionCompleted(ct)
+      }
+      ```
+   - Use the service to request review.
+      ``` cs
+      using ReviewService;
+
+      private readonly IReviewService<ReviewSettings> _reviewService;
+
+      public async Task OnCompletedImportantFlow(CancellationToken ct)
+      {
+          // Do Meaningful Task.
+
+          // Check if all conditions are satisfied and prompt for review if they are.
+          await _reviewService.TryRequestReview(ct);
+      }
+      ```
+
+   <img src="docs/review_prompt_android.gif" alt="Review Prompt Android" width="250">
+   <img src="docs/review_prompt_ios.gif" alt="Review Prompt iOS" width="250">
+
+### Persisting Review Settings
+
+`MemoryReviewSettingsSource` is great for automated testing but should not be the implementation of choice for real use-cases. Instead, you should create your own implementation that persists data on the device (so that review settings don't reset when you kill the app).
+
+``` cs
+using ReviewService;
+
+/// <summary>
+/// Storage implementation of <see cref="IReviewSettingsSource{TReviewSettings}"/>.
+/// </summary>
+/// <typeparam name="TReviewSettings">The type of the persisted object.</typeparam>
+public sealed class StorageReviewSettingsSource<TReviewSettings> : IReviewSettingsSource<TReviewSettings>
+    where TReviewSettings : ReviewSettings
+{
+    /// <inheritdoc/>
+    public Task<TReviewSettings> Read(CancellationToken ct)
+    {
+        // TODO: Return stored review settings.
+    }
+
+    /// <inheritdoc/>
+    public Task Write(CancellationToken ct, TReviewSettings reviewSettings)
+    {
+        // TODO: Update stored review settings.
+    }
+}
+```
+
+### Using Dependency Injection
+
+Here is a simple code that does dependency injection using `Microsoft.Extensions.DependencyInjection` and `Microsoft.Extensions.Hosting`.
+
+``` cs
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using ReviewService;
+
+var host = new HostBuilder()
+    .ConfigureServices(serviceCollection => serviceCollection
+        .AddSingleton<IReviewSettingsSource<ReviewSettingsCustom>, ReviewSettingsSource>()
+        .AddTransient(s => ReviewConditionsBuilder.Default<ReviewSettingsCustom>())
+        .AddSingleton<IReviewService<ReviewSettingsCustom>, ReviewService<ReviewSettingsCustom>>()
+    )
+    .Build();
+```
 
 ## Features
 
-{More details/listing of features of the project}
+The previous setup was pretty basic. Let's see what else we can do!
+
+### Default Review Conditions
```

Comment:
The feature isn't clear. There are at least 2 topics in this section and should be split.
The most important one is defining the conditions.
Another one would be customizing the tracking data, but I would only mention the review settings after the tracking of events.

Reply:
Features titles should start with a verb to better explain what you're doing with the SDK.